bionic/flow.py

```diff
@@ -114,22 +114,25 @@ def add_case(self, name, case_key, value):
     def clear_providers(self, names):
         state = self
 
-        # Remember the original protocol for each provider.
-        protocols_by_entity_name = {}
+        # Remember the original protocol and docstring for each provider.
+        attrs_by_entity_name = {}
         for name in names:
             if not state.has_provider(name):
                 continue
             provider = self.get_provider(name)
-            for res_name, res_protocol in zip(
-                    provider.attrs.names, provider.attrs.protocols):
-                protocols_by_entity_name[res_name] = res_protocol
+
+            for res_name, res_protocol, res_docstring in zip(
```

Comment:
What happens if `names` has multiple values?

Reply:
Addressing in incoming update (so we won't zip the docstring)!